commit 3f945585e96ea7c14fdc2fe640d4f56602973455
Author: Xie Xingguo <xie.xingguo@zte.com.cn>
Date:   Tue Dec 18 20:14:41 2018 +0800

    mgr/DaemonServer: allow merge only if merge participants are active+clean
    
    Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

diff --git a/src/mgr/DaemonServer.cc b/src/mgr/DaemonServer.cc
index a0700bb089e..08c5ef9751f 100644
--- a/src/mgr/DaemonServer.cc
+++ b/src/mgr/DaemonServer.cc
@@ -2370,8 +2370,8 @@ void DaemonServer::adjust_pgs()
 		         << (is_merge_source ? " (merge source)" : " (merge target)")
 		         << dendl;
 	        ok = false;
-	      } else if (!(q->second.state & (PG_STATE_ACTIVE |
-					      PG_STATE_CLEAN))) {
+	      } else if ((q->second.state & (PG_STATE_ACTIVE | PG_STATE_CLEAN)) !=
+                                            (PG_STATE_ACTIVE | PG_STATE_CLEAN)) {
 	        dout(10) << "pool " << i.first
 		         << " pg_num_target " << p.get_pg_num_target()
 		         << " pg_num " << p.get_pg_num()
@@ -2380,14 +2380,6 @@ void DaemonServer::adjust_pgs()
 		         << " not clean (" << pg_state_string(q->second.state)
 		         << ")" << dendl;
 	        ok = false;
-	      } else if (q->second.state & PG_STATE_REMAPPED) {
-	        dout(10) << "pool " << i.first
-		         << " pg_num_target " << p.get_pg_num_target()
-		         << " pg_num " << p.get_pg_num()
-                         << (is_merge_source ? " - merge source " : " - merge target ")
-                         << merge_participant
-		         << " remapped" << dendl;
-	        ok = false;
 	      }
             }
 
